sdk/core/azure-core/inc/azure/core/http/websockets/curl_websockets_transport.hpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief #Azure::Core::Http::WebSockets::WebSocketTransport implementation via CURL.
+ */
+
+#pragma once
+
+#include "azure/core/context.hpp"
+#include "azure/core/http/curl_transport.hpp"
+#include "azure/core/http/http.hpp"
+#include "azure/core/http/transport.hpp"
+#include "azure/core/http/websockets/websockets_transport.hpp"
+#include <memory>
+
+namespace Azure { namespace Core { namespace Http { namespace WebSockets {
+
+  struct CurlWebSocketTransportOptions : public Azure::Core::Http::CurlTransportOptions
+  {
+  };
+  /**
+   * @brief Concrete implementation of a WebSocket Transport that uses libcurl.
+   */
+  class CurlWebSocketTransport : public CurlTransport, public WebSocketTransport {
+  public:
+    /**
+     * @brief Construct a new CurlWebSocketTransport object.
+     *
+     * @param options Optional parameter to override the default options.
+     */
+    CurlWebSocketTransport(
+        CurlWebSocketTransportOptions const& options = CurlWebSocketTransportOptions())
+        : CurlTransport(options)
+    {
+    }
+
+    /**
+     * @brief Implements interface to send an HTTP Request and produce an HTTP RawResponse
+     *
+     * @param request an HTTP Request to be send.
+     * @param context A context to control the request lifetime.
+     *
+     * @return unique ptr to an HTTP RawResponse.
+     */
+    virtual std::unique_ptr<RawResponse> Send(Request& request, Context const& context) override;
+
+    /**
+     * @brief Indicates if the transports supports native websockets or not.
+     *
+     * @details For the CURL websocket transport, the transport does NOT support native websockets -
+     * it is the responsibility of the client of the WebSocketTransport to format WebSocket protocol
+     * elements.
+     */
+    virtual bool HasNativeWebsocketSupport() override { return false; }
+
+    /**
+     * @brief Gracefully closes the WebSocket, notifying the remote node of the close reason.
+     *
+     */
+    virtual void NativeClose() override;
+
+    // Native WebSocket support methods.
+    /**
+     * @brief Gracefully closes the WebSocket, notifying the remote node of the close reason.
+     *
+     * @details Not implemented for CURL websockets because CURL does not support native websockets.
+     *
+     */
```

Comment:
What is the uint16_t parameter?  Can you add @param info to the function docs.

Reply:
I can't add the parameter to this because CLANG gets REALLY annoyed at @param values that aren't in the method signature. I'll add some text.